updater/site/curse.py

```diff
@@ -23,40 +69,60 @@ class Curse(AbstractSite):
-            page = Curse.session.get(self.url)
-            page.raise_for_status()  # Raise an exception for HTTP errors
-            content_string = str(page.content)
-            main_zip_url, *classic_zip_url = re.findall(
-                r"cf-recentfiles-credits-wrapper ml-auto my-auto.+?href=\"(?P<download>.+?)\"",
-                content_string)
-            # if classic, choose the explicit "classic download" listed, or fall back to the only download available
-            zip_url = classic_zip_url[-1] if self.game_version is GameVersion.classic and classic_zip_url else main_zip_url
-            return f'https://www.curseforge.com{zip_url}/file'
+            latest_release = next(version for version in self.versions() if version.type >= self.addon_version)
+            return latest_release.download_link
         except Exception as e:
             raise self.download_error() from e
 
-    def get_latest_version(self):
+    def versions(self, *, page=1) -> Generator[CurseAddonVersion, None, None]:
+        """Yields a sequence of CurseAddonVersions corresponding to addon releases
+
+        Ordered descending in time, so the first version yielded is the most recent.
+        Will page through until exhausted.
+        """
+        if self.game_version == GameVersion.classic:
+            game_version_filter = '1738749986:67408'
+        elif self.game_version == GameVersion.retail:
+            game_version_filter = '1738749986:517'
+        else:  # Agnostic version
+            game_version_filter = ''
+        request_params = {'filter-game-version': game_version_filter, 'page': page}
         try:
-            page = Curse.session.get(self.url)
-            if page.status_code in [403, 503]:
-                logger.error("Curse is blocking requests because it thinks you are a bot... please try later.")
-            page.raise_for_status()  # Raise an exception for HTTP errors
-            content_string = str(page.content)
-            # the first one encountered will be the WoW retail version
-            main_version, *classic_version = re.findall(
-                r"cf-recentfiles.+?data-id=.+?data-name=\"(?P<version>.+?)\"",
-                content_string)
-            # if classic, choose the explicit "classic version" listed, or fall back to the only version available
-            return classic_version[-1] if self.game_version is GameVersion.classic and classic_version else main_version
+            p = Curse.session.get(f'{self.url}/files/all', params=request_params)
+            soup = bs4.BeautifulSoup(p.text, 'html.parser')
+            versions_table = soup.find('table', {'class': 'listing listing-project-file project-file-listing b-table b-table-a'})
+            # Header row consumed by _
+            _, *version_rows = versions_table.find_all('tr')
+            yield from (CurseAddonVersion.from_tr(row) for row in version_rows)
+
+            # determine if there are more pages of versions, recurse if so
```

Comment:
Given that the logic for pulling the latest version prefers official releases over beta versions over alpha versions -- the only situation where multiple requests to Curse should happen is if an addon has a full page of alpha+beta releases (looks like 25) when a user is tracking official releases.

My bet is that this would be rare enough not to worry about